File: src/python/IntegrationSDK/foundationallm/integration/models/analyze_request.py
"""
The AnalyzeRequest class encapsulates the information
required for the analyzer to analyze text content for PII.
"""
from typing import Optional
from pydantic import BaseModel

class AnalyzeRequest(BaseModel):
    """
    Request object to analyze text content for PII.
    Set anonymize to True to replace PII with a placeholder.
    """
    content: str
    anonymize: bool = False
    language: Optional[str] = "en"

File: src/dotnet/Gatekeeper/Models/Integration/AnalyzeRequest.cs
using System.Text.Json.Serialization;

namespace FoundationaLLM.Gatekeeper.Core.Models.Integration
{
    /// <summary>
    /// Encapsulates a text analysis request.
    /// </summary>
    public class AnalyzeRequest
    {
        /// <summary>
        /// The text to be analyzed.
        /// </summary>
        [JsonPropertyName("content")]
        public required string Content { get; set; }

        /// <summary>
        /// A flag used to tell if PII found by analysis should be anonymized.
        /// </summary>
        [JsonPropertyName("anonymize")]
        public required bool Anonymize {  get; set; }

        /// <summary>
        /// The language used to detect PII.
        /// </summary>
        [JsonPropertyName("language")]
        public string? Language {  get; set; }
    }
}
